controllers/dspipeline_controller.go

```diff
@@ -197,57 +200,203 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, nil
 	}
 
-	usingCustomDB := params.UsingExternalDB(dspa)
+	// Initialize conditions
+	conditions := []metav1.Condition{}
 
+	usingCustomDB := params.UsingExternalDB(dspa)
+	databaseReady := r.buildCondition("DatabaseReady", dspa.Generation, "DatabaseReady")
 	if !usingCustomDB {
+
 		err = r.ReconcileDatabase(ctx, dspa, params)
 		if err != nil {
+			databaseReady.Status = metav1.ConditionFalse
+			databaseReady.Reason = "DataBaseFailedToDeploy"
 			return ctrl.Result{}, err
 		}
+
+		if !r.isDeploymentAvailable(ctx, dspa, "mariadb") {
+			databaseReady.Status = metav1.ConditionFalse
+			databaseReady.Reason = "DataBaseFailedToDeploy"
```

Comment:
If deployment is not available, it does not mean it failed to deploy, it could just be progressing
can we have a more accurate reason for this?